packages/component/addon/index.js

```diff
@@ -201,7 +201,9 @@ export const layout = decoratorWithRequiredParams((target, params) => {
   );
   assert(
     `The @layout decorator must be provided a template, received: ${template}`,
-    (() => typeof template === 'object' && typeof template.indexOf === 'undefined')()
+    (() =>
+      typeof template === 'function' ||
+      (typeof template === 'object' && typeof template.indexOf === 'undefined'))()
```

Comment:
btw is there any reason this is an IIFE? If not, I'd remove that in this PR as well.

Reply:
No not particularly. I'd only use an IIFE in order to simplify the boolean logic, e.g.:

```js
assert(
  'whatever',
  (() => {
    if (condition) { return true; }
    if (otherThing) { return true; }
    return false;
  })()
```

It would probably make this specific case easier to think about, but isn't a giant deal.